query-engine/connector-test-kit-rs/query-engine-tests/tests/queries/simple/find_unique.rs

```diff
@@ -89,4 +89,141 @@ mod find_unique {
 
         Ok(())
     }
+
+    fn simple_uniq_idx_with_embedded() -> String {
+        indoc! {r#"
+        type Location {
+            address String
+        }
+
+        model A {
+            #id(id, Int, @id)
+            name String
+            location Location
+
+            @@unique([location.address])
+        }
+        "#}
+        .to_string()
+    }
+
+    #[connector_test(schema(simple_uniq_idx_with_embedded), only(MongoDb))]
+    async fn simple_embedded_type(runner: Runner) -> TestResult<()> {
+        run_query!(
+            runner,
+            indoc! {r#"mutation {
+            createManyA(data: [
+                {id: 1 name: "foo" location: {set: {address: "a"}}},
```

Comment:
That test is relevant to the unique index, not to the findUnique feature. I will assume the unique index insertion has tests regarding that but because this feature is related to the _query_ process under unique indices I won't cover those tests or cases here (or you will have to cover multiple cases in a query case).

Reply:
There are tests for write operations and unique indices already.